### PR TITLE
Add fecha v1.2.0/1.2.1 w/ npm auto-update

### LINKED ISSUE #11013: [Request] Fetcha - Javascript Date formatting and parsing
**Library name:** Fetcha
**Git repository url:** https://github.com/taylorhakes/fecha
**npm package name or url (if there is one):** fetcha
**License (List them all if it's multiple):** MIT License
**Official homepage:**
**Wanna say something? Leave message here:**
This library is basically `moment.js`, but without the manipulation part. As such, it's about 15% the size of Moment.

### PR DESCRIPTION
**Pull request by dakshshah96**
close #11013, cc @taylorhakes

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/taylorhakes/fecha
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/fecha
 * License and its reference:
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 20
   - Count of stars: 927
   - Count of forks: 50
 * NPM download stats (optional):
   - Downloads in the last day: 128
   - Downloads in the last week: 4,327
   - Downloads in the last month: 19,161

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by dakshshah96** [comment]
Hi @maruilian11, this is the new PR I've sent from the command line. Finally figured out how to make it work 👍

**Comment by dakshshah96** [comment]
@maruilian11 What should be the `basePath` here? Can't seem to figure it out.

**Comment by dakshshah96** [comment]
I've also managed to squash all my commits into one and sent them here from the command line. Hope everything's alright. 👍

File: ajax/libs/fecha/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "format",
+    "fecha",
+    "formatting"
+  ],
+  "npmName": "fecha",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "fecha.min.js"
```
Comment (maruilian11) [inline ajax/libs/fecha/package.json]: Please use `"fecha*` so that we can also fetch `fecha.js`. Thanks!

**Comment by maruilian11** [comment]
You can use `git rebase -i HEAD~2` to modify the origin commit and avoid reopening a new PR.

**Comment by maruilian11** [comment]
BTW, could you please also add `1.2.0` and `1.2.1` from git repo since npm package can't fetch these two versions.

**Comment by dakshshah96** [comment]
@maruilian11 Sure, I can add them. Both `fecha.js` and `fecha.min.js` should be included right?

**Comment by maruilian11** [comment]
Yes, thanks!

**Comment by dakshshah96** [comment]
@maruilian11 What should I do about this error message from the CI? `Static files for web hosting should not be executable! Please remove executable permission on the file(s) below`

**Comment by PeterDaveHello** [comment]
@dakshshah96 chmod -x

**Comment by dakshshah96** [comment]
@PeterDaveHello I did `chmod -x fecha.js` and similar for all four files in the two folders. The CI gives the same error again.

**Comment by PeterDaveHello** [comment]
@dakshshah96 you need to squash it to a single commit

**Comment by dakshshah96** [comment]
@PeterDaveHello On a side note, I can't really seem to grasp the syntax used to specify file rules in `npmFileMap`. Any resource you recommend so I can really understand what's happening?

**Comment by maruilian11** [comment]
@dakshshah96 We use regular expression in auto-update filemap. You can see https://github.com/isaacs/node-glob#glob-primer for more rules detail.
`*` can catch files in a certain directory, but we usually use it in `dist` or `built` folder because we usually need all file in them.
For this kind of case, we only need `fecha.js` and `fecha.min.js`, so we use `fecha*` to fetch both of them and keep the rule simple.

**Comment by dakshshah96** [comment]
Thanks @maruilian11, that clears up a lot of confusion! 💯

**Comment by maruilian11** [comment]
@dakshshah96 The commit msg should be `npm auto-update` rather than `git auto-update`.
Could you split the commit, one for package.json, and add old versions in another commit. Thanks!

**Comment by dakshshah96** [comment]
@maruilian11 I've fixed this PR. Hope everything's correct.

**Comment by dakshshah96** [comment]
@pvnr0082t What is the reason? The file structure seems fine for both the versions. I don't know.

**Comment by pvnr0082t** [comment]
>BTW, could you please also add 1.2.0 and 1.2.1 from git repo since npm package can't fetch these two versions.

Said by @maruilian11. You should mention it in the second commit message.

**Comment by dakshshah96** [comment]
@pvnr0082t Done 👍

**Comment by PeterDaveHello** [comment]
Wrong title men.

**Comment by PeterDaveHello** [comment]
@pvnr0082t as the file inconsistency problems in the two other PRs, have you checked the files in this one?

**Comment by PeterDaveHello** [comment]
Guys do we need `fecha.spec.js` in versions v1.x?

**Comment by dakshshah96** [comment]
@PeterDaveHello I don't think we have any need for `fecha.spec.js` as such files are used for writing unit test cases for a web app. These won't be needed by developers looking to use fecha online. 😄

**Comment by dakshshah96** [comment]
@PeterDaveHello @pvnr0082t Updated files to match release versions.

**Comment by PeterDaveHello** [comment]
@pvnr0082t @dakshshah96 so it'll be better to exclude in the rules for the old versions

**Comment by dakshshah96** [comment]
> @pvnr0082t @dakshshah96 so it'll be better to exclude in the rules for the old versions

@PeterDaveHello Didn't get what you meant by that.

**Comment by PeterDaveHello** [comment]
@dakshshah96 the current pattern will make the updater fetch the files (spec) from old versions

**Pull request merged**